test/connectivity/modules/NetworkController/LinuxNetworkInterfaceOfflineController.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace NetworkController
+{
+    using System;
+    using System.Net.NetworkInformation;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class LinuxNetworkInterfaceOfflineController : IController
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<LinuxNetworkInterfaceOfflineController>();
+        readonly string interfaceName;
+
+        public LinuxNetworkInterfaceOfflineController(string interfaceName)
+        {
+            this.interfaceName = interfaceName;
+        }
+
+        public string Description => "LinuxNetworkInterfaceOffline";
+
+        public Task<NetworkStatus> GetStatus(CancellationToken cs)
+        {
+            OperationalStatus status = this.GetNetworkInterfaceStatus();
+            switch (status)
+            {
+                case OperationalStatus.Up:
+                    return Task.FromResult(NetworkStatus.Default);
+                case OperationalStatus.Unknown:
+                    return Task.FromResult(NetworkStatus.Unknown);
+                default:
+                    return Task.FromResult(NetworkStatus.Restricted);
+            }
+        }
+
+        public Task<bool> SetStatus(NetworkStatus status, CancellationToken cs)
+        {
+            switch (status)
+            {
+                case NetworkStatus.Restricted:
+                    return this.Disable(cs);
+                case NetworkStatus.Default:
+                    return this.Enable(cs);
+                default:
+                    Log.LogDebug($"Status not set {status}");
+                    throw new NotSupportedException($"Set status '{status}' is not supported.");
+            }
+        }
+
+        async Task<bool> Disable(CancellationToken token)
+        {
+            try
+            {
+                string output = await CommandExecutor.Execute("ifconfig", $"{this.interfaceName} down", token);
```

Comment:
is output equivalent to exit code? if it is exit code, you should verify if exit code is zero for successful result. right?

Reply:
it's not the exit code, is the standard output. If exit code it != 0 it throws an exception from the CommandExecutor

---
In reply to: [359157653](https://github.com/Azure/iotedge/pull/2137#discussion_r359157653) [](ancestors = 359157653)